### PR TITLE
Added config options, shortcuts, integrated with Configuration Manage…

### PR DESCRIPTION
**Pull request by enkephalin07**
…r. Text better formated, character name included in subtitle. And other stuffs.

### CODE REVIEW

File: HSubs/HSubs.cs [hunk 1]
```diff
@@ -15,259 +15,480 @@
...
-		private IEnumerator Show_Coroutine(float expire, string subtitle)
-		{
-			currentLine = subtitle;
-			yield return new WaitForSeconds(expire);
-			currentLine = string.Empty;
-		}
+    [BepInPlugin(GUID: "org.bepinex.kk.hsubs", Name: "HSubs", Version: "3.0")]
+    public class HSubsPlugin : BaseUnityPlugin
+    {
+#if DEBUG
```
Comment (ManlyMarco) [inline HSubs/HSubs.cs:21]: Not used?

File: HSubs/HSubs.cs [hunk 2]
```diff
@@ -15,259 +15,480 @@
...
+            textOffset = new ConfigWrapper<float>("textOffset", this, 24);
+            outlineThickness = new ConfigWrapper<int>("outlineThickness", this, 2);
+            showMode = new ConfigWrapper<int>("showMode", this, 0);
+            updateOnStart = new ConfigWrapper<bool>("updateOnStart", this, true);
+            copyToClipboard = new ConfigWrapper<bool>("copyToClipboard", this, false);
+            copyJPLine = new ConfigWrapper<bool>("copyJPLine", this, false);
+        }
+
+//        public void Start()
+        private bool DoSomeShit()
```
Comment (ManlyMarco) [inline HSubs/HSubs.cs:125]: Name this init or something like that to be clearer

File: HSubs/HSubs.cs [hunk 3]
```diff
@@ -15,259 +15,480 @@
...
+            if (bodyBuilder.Length > 0)
+                row.Add(bodyBuilder.ToString());
+
+            if (row.Count > 0)
+                yield return row;
+        }
+
+        [HarmonyPostfix]
+        [HarmonyPatch(typeof(LoadVoice), "Play")]
+        public static void CatchVoice(LoadVoice __instance)
```
Comment (ManlyMarco) [inline HSubs/HSubs.cs:286]: I suggest splitting the harmony hooks into a separate class if possible.

File: HSubs/HSubs.cs [hunk 4]
```diff
@@ -15,259 +15,480 @@
...
-	}
-}
\ No newline at end of file
+        public void OnDestroy()
+        {
+            new PatchProcessor(harmony, typeof(HSubsPlugin), new HarmonyMethod(null)).Unpatch(HarmonyPatchType.All, GUID);
+        }
+    } 
+}
+/*
```
Comment (ManlyMarco) [inline HSubs/HSubs.cs:402]: No longer used? If there are no plans to bring this back then just remove it

**Review by ManlyMarco — COMMENTED** [review]
Code looks good to me, and it works well. Could just use some code cleanup

File: HSubs/HSubs.cs (hunk 1 above)
Comment (enkephalin07) [reply HSubs/HSubs.cs:21]: It's still used; some reorganized, some revised and renamed to identify the changes.

File: HSubs/HSubs.cs (hunk 2 above)
Comment (enkephalin07) [reply HSubs/HSubs.cs:125]: It'll be clearer after I break it down to separate subclasses that won't need to initialize at startup.

File: HSubs/HSubs.cs (hunk 4 above)
Comment (enkephalin07) [reply HSubs/HSubs.cs:402]: These closing comments are reference/roadmap for future development. Most of it, potentially all of it, will be integrated in some form.

**Pull request closed without merging**